File: 3j/cord/zenki/kadai-6/17-22059-來間空-kadai-6.c
#include<stdio.h>
double saidie(double x, double y, double z){
	double max = x;
	if(y > max){
		max = y;
	}
	else if(z > max){
		max = z;
	}
	return max;
}

double saisho(double x, double y, double z){
	double min = x;
	if(y < min){
		min = y;
	}
	else if(z < min){
		min = z;
	}
	return min;
}
double ghokei(double x, double y, double z){
	double sum = 0;
	sum = x + y + z;
	return sum;
}

double heykin(double x, double y, double z){
	double ave = 0;
	ave = ghokei(x,y,z)/3;
	return ave;
}

int main(void){
	double x = 0;
	double y = 0;
	double z = 0;
	int n = 0;
	printf("１つ目の数値を入力:");
	scanf("%lf",&x);
	printf("２つ目の数値入力：");
	scanf("%lf",&y);
	printf("３つ目の数値入力：");
	scanf("%lf",&z);
	printf("処理を入力。0:最大、1:最小、2:合計、3:平均:");
	scanf("%d",&n);
	if(n == 0){
		printf("%lf\n",saidie(x, y, z));
	}else if(n == 1){		
		printf("%lf\n",saisho(x, y, z));
	}else if(n == 2){
		printf("%lf\n",ghokei(x, y, z));
	}else if(n == 3){
		printf("%lf\n",heykin(x, y, z));

	}else{
		printf("入力が正しくありません\n");
	}

	return 0;
}

File: 3j/cord/zenki/kadai-6/17-22059-來間空-kadai-6.py


def find_average(numbers):
    total = sum(numbers)
    average = total / len(numbers)
    return average

def find_maximum(numbers):
    return max(numbers)

def find_minimum(numbers):
    return min(numbers)

def find_sum(numbers):
    return sum(numbers)

def main():
    print("3つの数値(小数点以下を含む)を入力")
    numbers = []
    for i in range(3):
        number = float(input("数値を入力: "))
        numbers.append(number)
    
    choice = input("最大値'max'、最小値'min'、合計値'sum'、平均値'average' を入力: ")

    if choice == 'max':
        result = find_maximum(numbers)
        print("最大値は", result)
    elif choice == 'min':
        result = find_minimum(numbers)
        print("最小値は", result)
    elif choice == 'sum':
        result = find_sum(numbers)
        print("合計値は", result)
    elif choice == 'average':
        result = find_average(numbers)
        print("平均値は", result)
    else:
        print("無効な選択")

if __name__ == "__main__":
    main()
